deepchem/models/tensorgraph/models/seqtoseq.py

```diff
@@ -0,0 +1,267 @@
+               dropout=0.0,
+               **kwargs):
+    """Construct a SeqToSeq model.
+
+    In addition to the following arguments, this class also accepts all the keyword arguments
+    from TensorGraph.
+
+    Parameters
+    ----------
+    input_tokens: list
+      a list of all tokens that may appear in input sequences
+    output_tokens: list
+      a list of all tokens that may appear in output sequences
+    max_output_length: int
+      the maximum length of output sequence that may be generated
+    encoder_layers: int
+      the number of recurrent layers in the encoder
+    decoder_layers: int
+      the number of recurrent layers in the decoder
+    embedding_dimension: int
+      the width of the embedding vector.  This also is the width of all
+      recurrent layers.
+    dropout: float
+      the dropout probability to use during training
+    """
+    super(SeqToSeq, self).__init__(
+        use_queue=False, **kwargs)  # TODO can we make it work with the queue?
+    if SeqToSeq.sequence_end not in input_tokens:
+      input_tokens = input_tokens + [SeqToSeq.sequence_end]
+    if SeqToSeq.sequence_end not in output_tokens:
+      output_tokens = output_tokens + [SeqToSeq.sequence_end]
+    self._input_tokens = input_tokens
+    self._output_tokens = output_tokens
+    self._input_dict = dict((x, i) for i, x in enumerate(input_tokens))
+    self._output_dict = dict((x, i) for i, x in enumerate(output_tokens))
+    self._max_output_length = max_output_length
+    self._features = layers.Feature(shape=(None, None, len(input_tokens)))
+    self._labels = layers.Label(shape=(None, None, len(output_tokens)))
+    self._gather_indices = layers.Feature(shape=(None, 2), dtype=tf.int32)
+    prev_layer = self._features
+    for i in range(encoder_layers):
+      if dropout > 0.0:
+        prev_layer = layers.Dropout(dropout, in_layers=prev_layer)
+      prev_layer = layers.GRU(
+          embedding_dimension, self.batch_size, in_layers=prev_layer)
+    prev_layer = layers.Gather(in_layers=[prev_layer, self._gather_indices])
+    self.embedding = prev_layer
+    prev_layer = layers.Repeat(max_output_length, in_layers=prev_layer)
+    for i in range(decoder_layers):
+      if dropout > 0.0:
+        prev_layer = layers.Dropout(dropout, in_layers=prev_layer)
+      prev_layer = layers.GRU(
+          embedding_dimension, self.batch_size, in_layers=prev_layer)
+    output_layer = layers.Dense(
+        len(output_tokens), in_layers=prev_layer, activation_fn=tf.nn.softmax)
+    self.add_output(output_layer)
+    prob = layers.ReduceSum(
+        layers.Multiply([output_layer, self._labels]), axis=2)
+    log_prob = layers.Log(
+        layers.Add([prob, layers.Constant(np.finfo(np.float32).eps)]))
+    objective = layers.ReduceMean(layers.ReduceSum(log_prob, axis=1))
+    loss = layers.Multiply([objective, layers.Constant(-1)])
+    self.set_loss(loss)
+    self.output = output_layer
+
+  def fit_sequences(self,
```

Comment:
We might not want a new name here; we could just reuse `fit_generator` since the only meaningful inputs for `Seq2Seq` is sequences.

Reply:
I'd like to keep open the option of calling `fit_generator()`.  Note that although the two methods both taking generators, they're very different.  This one takes a generator that produces pairs of sequences.  The other takes a generator that produces feed dicts.